internal/vault/secrets/azure/azure_static_role_resource.go

```diff
@@ -0,0 +1,361 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/base"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/client"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/errutil"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/model"
+)
+
+const staticRolesAffix = "static-roles"
+
+var idRe = regexp.MustCompile(`^([^/]+)/` + staticRolesAffix + `/([^/]+)$`)
+
+// Ensure the implementation satisfies the resource.ResourceWithConfigure interface
+var _ resource.ResourceWithConfigure = &AzureSecretsStaticRoleResource{}
+
+// NewAzureStaticRoleResource returns the implementation for this resource to be
+// imported by the Terraform Plugin Framework provider
+func NewAzureStaticRoleResource() resource.Resource { return &AzureSecretsStaticRoleResource{} }
+
+// AzureSecretsStaticRoleResource implements the methods that define this resource
+type AzureSecretsStaticRoleResource struct {
+	base.ResourceWithConfigure
+	base.WithImportByID
+}
+
+// AzureStaticRoleModel describes the Terraform resource data model to match the
+// resource schema.
+type AzureStaticRoleModel struct {
+	base.BaseModelLegacy
+
+	Backend             types.String `tfsdk:"backend"`
+	Role                types.String `tfsdk:"role"`
+	ApplicationObjectID types.String `tfsdk:"application_object_id"`
+	TTL                 types.Int64  `tfsdk:"ttl"`
+	Metadata            types.Map    `tfsdk:"metadata"`
+	SecretID            types.String `tfsdk:"secret_id"`
+	ClientSecret        types.String `tfsdk:"client_secret"`
+	Expiration          types.String `tfsdk:"expiration"`
+	SkipImportRotation  types.Bool   `tfsdk:"skip_import_rotation"`
+}
+
+// AzureStaticRoleAPIModel describes the Vault API data model.
+type AzureStaticRoleAPIModel struct {
+	ApplicationObjectID string            `json:"application_object_id" mapstructure:"application_object_id"`
+	TTL                 any               `json:"ttl" mapstructure:"ttl"`
+	Metadata            map[string]string `json:"metadata" mapstructure:"metadata"`
+}
+
+func (r *AzureSecretsStaticRoleResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_azure_secret_backend_static_role"
+}
+
+func (r *AzureSecretsStaticRoleResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			consts.FieldBackend: schema.StringAttribute{
+				MarkdownDescription: "The path where the Azure secrets backend is mounted.",
+				Optional:            true,
```

Comment:
In the argument reference in the doc you have this field as "(Required)", so I think either this or the doc needs to be adjusted.

Reply:
Good catch. I'll make this required here as well to stay consistent with others.